### PR TITLE
Update setup files and add new files

### PR DESCRIPTION
**Pull request by skupr-anaconda**
Actions:
1. Add [.editorconfig](https://github.com/anaconda-distribution/anaconda-linter/pull/90/files#diff-0947e2727d6bad8cd0ac4122f5314bb5b04e337393075bc4b5ef143b17fcbd5b)
2. Add [.github/workflows/ISSUE_TEMPLATE.md](https://github.com/anaconda-distribution/anaconda-linter/pull/90/files#diff-4ec6e84c871bcc442328d88eec1190974cd49fe29c5decf76866771beb46fe63)
3. Update `.gitignore`: sort lines, add new entries.
4. Add [AUTHORS.rst](https://github.com/anaconda-distribution/anaconda-linter/pull/90/files#diff-4e28079e8f9a4e208ee815c55aa8f1d87ca0698da61f044486ca60de48e7a525)
5. Add [CHANGELOG.md](https://github.com/anaconda-distribution/anaconda-linter/pull/90/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed)
6. Add [CONTRIBUTING.rst](https://github.com/anaconda-distribution/anaconda-linter/pull/90/files#diff-76e7ca2f2843ac5f659b45382768c1b64bf8704e8d3c6189c4e3623f5d465bc2)
7. Add [MANIFEST.in](https://github.com/anaconda-distribution/anaconda-linter/pull/90/files#diff-41d5a52589e0480be9c099d2bba7a8135b8b0d71bcbb8df3582a8df1c2295003)
8. Update [Makefile](https://github.com/anaconda-distribution/anaconda-linter/pull/90/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52): add `.PHONY`, `BROWSER`, `clean`, `coverage`, `lint`, `release`, `dist`, `install`.
9. Update [README.md](https://github.com/anaconda-distribution/anaconda-linter/pull/90/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5): move the content of `__long_description__` from `anaconda_lint.__init__.py` to `README.md`, add `license` entry
10. Add `__name__` and `__copyright__` to `anaconda_lint.__init__.py`
11. Improve and extend [setup.py](https://github.com/anaconda-distribution/anaconda-linter/pull/90/files#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7)

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
I think for installation we should use `pip install -e .` instead of `python setup.py install` like here https://github.com/anaconda-distribution/anaconda-linter/blob/f017e96aa2610d64de830d5280d7630c28702957/Makefile#L13

**Review by razzlestorm — APPROVED** [review]
This looks great, how sure are we that this isn't going to impact the feedstock/conda packaging of the linter (the setup.py specifically)?

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+Pull Request Guidelines
+-----------------------
+
+Before you submit a pull request, check that it meets these guidelines:
+
+1. The pull request should include tests.
+2. If the pull request adds functionality, the docs should be updated. Put
+   your new functionality into a function with a docstring, and add the
+   feature to the list in README.rst.
+3. The pull request should work for Python 3.5, 3.6, 3.7 and 3.8, and for PyPy.
```
Comment (marcoesters) [inline CONTRIBUTING.rst:105]: We don't support 3.5 and 3.6 anymore, and `anaconda-linter` is not compatible with 3.7 either.

File: README.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 # Anaconda Linter
 
-This package is currently a very rough draft of a pure Python linter specifically to make sure
+Anaconda Linter is a utility to validate that a recipe for a conda package
+will render correctly, and has been filled out with the correct information.
```
Comment (marcoesters) [inline README.md:4]:
```suggestion
will render correctly.
```

**Comment by marcoesters** [comment]
> This looks great, how sure are we that this isn't going to impact the feedstock/conda packaging of the linter (the setup.py specifically)?

The recipe will have to be changed, yes.

File: README.md (hunk 2 above)
Comment (marcoesters) [reply README.md:4]: Or rather add something along the lines of "contains all required information" if that's what you intended to write.

File: CONTRIBUTING.rst (hunk 1 above)
Comment (skupr-anaconda) [reply CONTRIBUTING.rst]: I've updated the text

**Comment by danpetry** [comment]
@skupr-anaconda could you please update the feedstock as well? Otherwise it'll break the prefect environment.

**Comment by skupr-anaconda** [comment]
@danpetry Sure, but we can do it after releasing a new version 0.0.3, right?

**Comment by danpetry** [comment]
ah right, yes true

**Pull request merged**